foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_7_0/EnableHabushuBuildCacheMigration.java

```diff
@@ -0,0 +1,116 @@
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.*;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringReader;
+
+
+/**
+ * This migration adds a new build directory tag to the build node of the project's pom.xml file.
+ */
+public class EnableHabushuBuildCacheMigration extends AbstractAissembleMigration {
+    private static final String BUILD_NODE_PATH = "/project/build";
+    private static final String PACKAGING_NODE_PATH = "/project/packaging";
+    private static final String BUILD_DIR_NODE_PATH = "/project/build/directory";
+
+    // XSL string to strip extra empty newlines from the migrated file-- negates a yucky side effect of the transformer
+    private static final String xsl =
+            "<xsl:stylesheet version=\"1.0\" xmlns:xsl=\"http://www.w3.org/1999/XSL/Transform\">\n" +
+            "  <xsl:output indent=\"yes\"/>\n" +
+            "  <xsl:strip-space elements=\"*\"/>\n" +
+            "\n" +
+            "  <xsl:template match=\"@*|node()\">\n" +
+            "    <xsl:copy>\n" +
+            "      <xsl:apply-templates select=\"@*|node()\"/>\n" +
+            "    </xsl:copy>\n" +
+            "  </xsl:template>\n" +
+            "\n" +
+            "</xsl:stylesheet>";
+
+    /**
+     * Retrieves the xml node at the specified path
+     *
+     * @param file The xml file to parse
+     * @param expression The path to the target node
+     * @return The retrieved node, or null if the node at the specified path does not exist.
+     */
+    public static Node getNode(File file, String expression) {
+        DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder;
+        try {
+            builder = builderFactory.newDocumentBuilder();
+            Document xmlDocument = builder.parse(file);
+            XPath xPath = XPathFactory.newInstance().newXPath();
+            return (Node) xPath.compile(expression).evaluate(xmlDocument, XPathConstants.NODE);
+        } catch (ParserConfigurationException | XPathExpressionException | IOException | SAXException e) {
+            throw new RuntimeException(e);
```

Comment:
S: We should not throw generic exception - better to use BatonException with a message in this case (https://rules.sonarsource.com/java/type/Code%20Smell/RSPEC-112/)

```suggestion
            throw new BatonException("Could not find specified node for expression: " + expression, e);
```

Reply:
Made a change here-- Took it a slightly different and more generic direction than the suggestion based on the most likely cause of failure.